e2e/operatorutil/podmonitoringutil.go

```diff
@@ -118,6 +118,41 @@ func isPodMonitoringScrapeEndpointSuccess(status *monitoringv1.ScrapeEndpointSta
 	return errors.Join(errs...)
 }
 
+func isPodMonitoringScrapeEndpointFailure(status *monitoringv1.ScrapeEndpointStatus, expectedFn func(message string) error) error {
+	var errs []error
+	if status.UnhealthyTargets == 0 {
+		errs = append(errs, errors.New("expected no healthy targets"))
+	}
+	if status.CollectorsFraction == "0" {
+		errs = append(errs, fmt.Errorf("expected collectors fraction to be 0 but found: %s", status.CollectorsFraction))
+	}
+	if len(status.SampleGroups) == 0 {
+		errs = append(errs, errors.New("missing sample groups"))
```

Comment:
Perhaps returning early would be a bit more readabilty (less left indents below)